### ISSUE TITLE
Skybox corners visible with Vulkan and low screen resolutions

### ISSUE THREAD

**Issue by rg3** [yquake2/yquake2#639]
When screen resolutions are low (see the attached 320x240 screenshot), the skybox corners appear to be visible. I think the problem also happens with OpenGL renderers, but it's not as noticeable there.

![q2_0003](https://user-images.githubusercontent.com/53487/103041493-d8c24580-4576-11eb-8eb1-391267af8276.jpg)

**Comment by rg3** [comment]
Forgot to mention this happens under Fedora 33 with NVIDIA proprietary drivers 455.45.01 using a GTX 1070, just in case it turns out to be a Vulkan driver problem.

**Comment by Yamagi** [comment]
The same happens with RADV on my 5700XT. So no driver problem.

**Comment by rg3** [comment]
I've been taking a look at this today and I've found out using samplers that clamp texture coordinates to the edge when drawing the skybox solves this visual problem. This is probably related to texture coordinates in the skybox. The Vulkan renderer uses several samplers for different settings, so my patch involves creating a second set of samplers that are identical to the existing ones except that the address mode is set to clamp to edge, and when drawing the skybox we can use that second set of samplers. I can submit that in a pull request if you're interested.

![q2_0005](https://user-images.githubusercontent.com/53487/104131056-61f9ba80-5374-11eb-9be5-ef61fb5bbf71.jpg)

That said, once I solved that problem and while I was taking a closer look at the sky in detail I noticed some flickering black pixels along the edges of skyboxes. It's usually a single pixel or two and only in some frames. While debugging that second problem with renderdoc, I found out the way the skybox is clipped results in geometry that's not watertight. Adjacent triangles don't share the same vertex coordinates, which means under some circumstances the pixel sample coordinates fall outside either triangle and are rendered black. As always, with higher resolutions you normally can't see this but with lower resolutions these black pixels are larger and more noticeable. I'll open a separate issue for this second problem, but I don't think I have time for a more detailed investigation into ClipSkyPolygon and friends.

![q2_0004](https://user-images.githubusercontent.com/53487/104130922-bb151e80-5373-11eb-8e49-da846e1b0f97.jpg)
